stock_location_tank/models/stock_picking.py

```diff
@@ -0,0 +1,165 @@
+                production = rec.env["mrp.production"].create(
+                    {
+                        "product_id": product.id,  # Required
+                        "product_qty": quantity_to_prod,  # Required
+                        "product_uom_id": product.uom_id.id,  # Required
+                        "picking_type_id": mrp_operation_type.id,  # Required
+                        "location_src_id": mrp_operation_type.default_location_src_id.id,  # Required
+                        "location_dest_id": location_dest.id,  # Required
+                        "picking_id": rec.id,
+                        "move_raw_ids": [
+                            (
+                                0,
+                                0,
+                                {
+                                    "name": product.name,  # Required
+                                    "product_id": product.id,  # Required
+                                    "location_id": mrp_operation_type.default_location_src_id.id,  # Required
+                                    "location_dest_id": location_dest.id,  # Required
+                                    "product_uom": product.uom_id.id,  # Required
+                                    "product_uom_qty": quantity_to_prod,  # Required
+                                },
+                            )
+                        ],
+                    }
+                )
+                production._onchange_move_finished_product()  # depends('product_id')
+                production._onchange_move_finished()  # depends('product_qty', 'product_uom_id')
+                production._onchange_location_dest()  # depends('location_dest_id')
+                production.action_confirm()
+                lot_lmp = rec.env["stock.production.lot"].create(
+                    {
+                        "name": location_dest.sequence_id._next(),
+                        "product_id": product.id,
+                        "product_qty": qty_done,
+                        "product_uom_id": product.uom_id.id,
+                    }
+                )
+                vals = []
+                for component in component_quant:
+                    vals.append(
+                        (
+                            0,
+                            0,
+                            {
+                                "lot_id": component.lot_id.id,
+                                "product_id": product.id,
+                                "qty_done": component.quantity,
+                                "product_uom_id": product.uom_id.id,
+                                "location_id": location_dest.id,
+                                "location_dest_id": product.with_company(
+                                    rec.company_id
+                                ).property_stock_production.id,
+                            },
```

Comment:
This should be inside a hook `_prepare_production_move_line_values` and it should return just a dictionary without the "0's". The "0's" should be added below at the last moment just before the assignation.